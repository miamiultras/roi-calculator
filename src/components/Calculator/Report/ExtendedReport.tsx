import React from 'react';

import Paragraph from '../../Paragraph/Paragraph';
import BackButton from '../../BackButton/BackButton';
import Table from '../Table/Table';
import { Service } from '../../../interfaces/service.interface';

const ExtendedReport = ({ extendedData }: { extendedData: Service[] }) => (
  <div className='container'>
    <h1>Live PPV Event ROI Calculator</h1>
    <Paragraph>
      Dear User, based on your entries we did the math for you:
   </Paragraph>
    <Table data={extendedData} />
    <BackButton name='start again' to='' />
  </div>
);

export default ExtendedReport;
