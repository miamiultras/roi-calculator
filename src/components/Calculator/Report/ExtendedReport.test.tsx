import * as React from 'react';
import { shallow } from 'enzyme';
import ExtendedReport from './ExtendedReport';

function setup() {
  const props = {};
  const wrapper = shallow(<ExtendedReport extendedData={[]} />);
  return {
    wrapper,
    props,
  };
}

describe('ExtendedReport component', () => {
  it('should match snapshot if no change was intended', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });
});
