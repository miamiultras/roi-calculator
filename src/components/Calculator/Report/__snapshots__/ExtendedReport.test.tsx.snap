// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExtendedReport component should match snapshot if no change was intended 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExtendedReport
    extendedData={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Live PPV Event ROI Calculator
        </h1>,
        <Paragraph>
          Dear User, based on your entries we did the math for you:
        </Paragraph>,
        <Table
          data={Array []}
        />,
        <BackButton
          name="start again"
          to=""
        />,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Live PPV Event ROI Calculator",
        },
        "ref": null,
        "rendered": "Live PPV Event ROI Calculator",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Dear User, based on your entries we did the math for you:",
        },
        "ref": null,
        "rendered": "Dear User, based on your entries we did the math for you:",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "data": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "name": "start again",
          "to": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Live PPV Event ROI Calculator
          </h1>,
          <Paragraph>
            Dear User, based on your entries we did the math for you:
          </Paragraph>,
          <Table
            data={Array []}
          />,
          <BackButton
            name="start again"
            to=""
          />,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Live PPV Event ROI Calculator",
          },
          "ref": null,
          "rendered": "Live PPV Event ROI Calculator",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Dear User, based on your entries we did the math for you:",
          },
          "ref": null,
          "rendered": "Dear User, based on your entries we did the math for you:",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "data": Array [],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "name": "start again",
            "to": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
